src/core/thread/mlr_manager.cpp

```diff
@@ -529,6 +588,84 @@ void MlrManager::AppendToUniqueAddressList(Ip6::Address (&aAddresses)[kIPv6Addre
     return;
 }
 
+bool MlrManager::AddressListContains(const Ip6::Address *aAddressList,
+                                     uint8_t             aAddressListSize,
+                                     const Ip6::Address &aAddress)
+{
+    bool contains = false;
+
+    VerifyOrExit(aAddressListSize > 0, contains = true);
```

Comment:
add comment like below to avoid confusion, for  better readability as well?

> For non `kMlrSuccess` response which doesn't have IPv6AddressesTlv,  we take all MAs in registering state as failed address .